models/optional_modules/core_web_vitals/snowflake/snowplow_web_vital_measurements.sql

```diff
@@ -0,0 +1,74 @@
+    grouping_id(cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }}), device_class) as id_period_and_device,
+    grouping_id(geo_country) as id_country,
+    grouping_id(geo_country, device_class) as id_country_and_device,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by lcp) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by fid) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by cls) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by ttfb) as ttfb_{{ var('snowplow__cwv_percentile') }}p
+  from {{ ref('snowplow_web_vitals') }}
+
+  where cast(derived_tstamp as date) >= {{ dateadd('day', '-'+var('snowplow__cwv_days_to_measure')|string, date_trunc('day', snowplow_utils.current_timestamp_in_utc())) }}
+
+  group by grouping sets ((), (page_url, device_class), (device_class), (cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }})), (cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }}), device_class), (geo_country), (geo_country, device_class))
+
+)
+
+, measurement_type as (
+
+  select
+    *,
+    case when id_url_and_device <> 0 and id_device <> 0 and id_period <> 0 and id_period_and_device <> 0 and id_country <> 0 and id_country_and_device <> 0 then 'overall'
+       when id_url_and_device = 0 then 'by_url_and_device'
+       when id_period_and_device = 0 then 'by_day_and_device'
+       when id_country_and_device = 0 then 'by_country_and_device'
+       when id_country = 0 then 'by_country'
+       when id_device = 0 then 'by_device'
+       when id_period = 0 then 'by_day'
+       end as measurement_type,
+   {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
+  from measurements
+)
+
+select
+  m.measurement_type,
+  coalesce(m.page_url, 'all') as page_url,
+  coalesce(m.device_class, 'all') as device_class,
+  coalesce(m.geo_country, 'all') as geo_country,
+  coalesce(g.name, 'all') as country,
+  coalesce(time_period, 'last {{var("snowplow__cwv_days_to_measure")|string }} days') as time_period,
+  page_view_count,
+  ceil(lcp_{{ var('snowplow__cwv_percentile') }}p, 3) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+  ceil(fid_{{ var('snowplow__cwv_percentile') }}p, 3) as fid_{{ var('snowplow__cwv_percentile') }}p,
+  ceil(cls_{{ var('snowplow__cwv_percentile') }}p, 3) as cls_{{ var('snowplow__cwv_percentile') }}p,
+  ceil(ttfb_{{ var('snowplow__cwv_percentile') }}p, 3) as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+  lcp_result,
+  fid_result,
+  cls_result,
+  ttfb_result,
+  case when lcp_result = 'good' and fid_result = 'good' and cls_result = 'good' then 1 else 0 end passed
+
```

Comment:
Probably just make sure in the macro docs that if they overwrite this macro it still needs to output 'good' for a pass result?

Reply:
Yeah, I planned to highlight this as they may want to include the null ones in the pass results